File: 10684 The jackpot/solution.c
#include <stdio.h>

int main() {
    int N;
    while(scanf("%d", &N) && N != 0) {
        int max = 0, currMax = 0, temp;
        while(N--) {
            scanf("%d", &temp);
            currMax = temp > currMax + temp ? temp : currMax + temp;
            max = max > currMax ? max : currMax;
        }
        if(max > 0)
            printf("The maximum winning streak is %d.\n", max);
        else
            puts("Losing streak.");
    }
	return 0;
}

File: 10684 The jackpot/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
        int N;
        StringBuilder output = new StringBuilder();
    	while(st.nextToken() == StreamTokenizer.TT_NUMBER && (N = (int)st.nval) != 0) {
    	    int max = 0, currMax = 0;
    	    while(N-- > 0) {
    	        st.nextToken();
    	        int temp = (int)st.nval;
    	        currMax = Math.max(temp, currMax + temp);
    	        max = Math.max(max, currMax);
    	    }
    	    if(max > 0)
    	        output.append("The maximum winning streak is ").append(max).append(".\n");
    	    else
    	        output.append("Losing streak.\n");
    	}
        System.out.print(output);
	}
}
